services/QuillLMS/client/app/bundles/PremiumHub/components/dataExportTableAndFields.tsx

```diff
@@ -0,0 +1,234 @@
+
+  function getData() {
+    initializePusher()
+
+    const searchParams = {
+      query: queryKey,
+      timeframe: selectedTimeframe,
+      timeframe_custom_start: customTimeframeStart,
+      timeframe_custom_end: customTimeframeEnd,
+      school_ids: selectedSchoolIds,
+      teacher_ids: selectedTeacherIds,
+      classroom_ids: selectedClassroomIds,
+      grades: selectedGrades
+    }
+
+    const requestUrl = queryString.stringifyUrl({ url: '/snapshots/data_export', query: searchParams }, { arrayFormat: 'bracket' })
+
+    requestGet(`${requestUrl}`, (body) => {
+      if (!body.hasOwnProperty('results')) {
+        setLoading(true)
+      } else {
+        const { results, } = body
+        // We consider `null` to be a lack of data, so if the result is `[]` we need to explicitly `setData(null)`
+        const data = results.length > 0 ? results : null
+        setData(data)
+        setLoading(false)
+      }
+    })
+  }
+
+  function initializePusher() {
+    const pusher = new Pusher(process.env.PUSHER_KEY, { encrypted: true, });
+    const channel = pusher.subscribe(String(adminId));
+    channel.bind('data-export-cached', (body) => {
+      const { message, } = body
+
+      const queryKeysAreEqual = message.query === queryKey
+      const timeframesAreEqual = message.timeframe === selectedTimeframe
+      const schoolIdsAreEqual = unorderedArraysAreEqual(message.school_ids, selectedSchoolIds.map(id => String(id)))
+      const teacherIdsAreEqual = unorderedArraysAreEqual(message.teacher_ids, selectedTeacherIds.map(id => String(id)))
+      const classroomIdsAreEqual = unorderedArraysAreEqual(message.classroom_ids, selectedClassroomIds.map(id => String(id)))
+      const gradesAreEqual = unorderedArraysAreEqual(message.grades, selectedGrades.map(grade => String(grade))) || (!message.grades && !selectedGrades.length)
+
+      if (queryKeysAreEqual && timeframesAreEqual && schoolIdsAreEqual && gradesAreEqual && teacherIdsAreEqual && classroomIdsAreEqual) {
+        getData()
+      }
+    });
+  };
+
+  function toggleCheckbox(e) {
+    e.preventDefault();
+    const fieldLabel = e.currentTarget.id;
+    const checked = fields[fieldLabel].checked;
+    const setterFunction = fields[fieldLabel].setterFunction
+    setterFunction(!checked);
+  }
+
+  function renderCheckboxes() {
+    return Object.keys(fields).map((fieldLabel, i) => {
+      // skip Student Name because we want this field to always be present
+      if (i === 0) { return }
+      const selected = fields[fieldLabel].checked;
+      const selectedClass = selected ? "selected" : "unselected";
+      const checkboxImg = selected ? <img alt="check" src={smallWhiteCheckIcon.src} /> : ""
+      return (
+        <div className="checkbox-container">
+          <div className={`quill-checkbox ${selectedClass}`} id={fieldLabel} onClick={toggleCheckbox}>{checkboxImg}</div>
```

Comment:
I don't think we've been totally consistent about this, but in the interests of accessibility I think it's best to use buttons with `aria-labels` for the checkboxes -- example here: https://github.com/empirical-org/Empirical-Core/blob/develop/services/QuillLMS/client/app/bundles/Teacher/components/update_assigned_students/unassign_warning_modal.tsx#L8.

Reply:
that makes sense-- thanks for the code reference! I'll update